plugins/fileservices/fileservices.cpp

```diff
@@ -289,12 +289,49 @@ static const char *getAccessibleEspServerURL(const char *param, IConstWorkUnit *
     if (param&&*param)
         return param;
 
-#ifdef _CONTAINERIZED
-    UNIMPLEMENTED_X("CONTAINERIZED(getAccessibleEspServerURL)");
-#else
     CriticalBlock b(espURLcrit);
     if (isUrlListEmpty())
     {
+#ifdef _CONTAINERIZED
+        const char *defaultEspService = getComponentConfigSP()->queryProp("@defaultEsp");
+        Owned<IPropertyTree> globalConfig = getGlobalConfig();
+        if (isEmptyString(defaultEspService))
+            defaultEspService = globalConfig->queryProp("@defaultEsp");
+
+        const IPropertyTree *match = nullptr;
+        if (!isEmptyString(defaultEspService))
+        {
+            VStringBuffer service("services[@name='%s']", defaultEspService);
+            match = globalConfig->queryPropTree(service.str());
+        }
+
+        if (!match)
+        {
+            // Look for 'eclservices' esp service, fallback to 'eclwatch' service.
+            Owned<IPropertyTreeIterator> iter = globalConfig->getElements("services");
+            ForEach(*iter)
+            {
+                const char *type = iter->query().queryProp("@type");
```

Comment:
Should this always be restricted to non-public services?  (i.e. ignore public services).

I'm not convinced we should be looking at defaultEsp either.  @afishbeck @richardkchapman what do you think?

Reply:
I do think we should always have a non-public eclservices for internal webservice calls.. and perhaps things like fileservices that call out to those services should stop working if one isn't present.  Workunit tokens (which I think will change, but still be used) won't work otherwise, and nor will mtls.

If defaultEsp were just used to specify which eclservices to use it would be fine although not sure why there would be more than one.  If it said that local webservice calls should be sent somewhere else that could be dangerous.  Perhaps we could do without defaultEsp for now, but re-add something similar if actually needed in the future?

Reply:
I thought the whole purpose of defaultESP was to be returned from functions like this!

Reply:
We could keep it as a way of disambiguating the eclservices if there is more than one, but should we require it to be non-public?  I really think these calls should use mtls when they can, and workunit tokens when necessary.. which both imply non-public.

Reply:
@richardkchapman As far as I can see defaultEsp was introduced in your change 
https://github.com/hpcc-systems/HPCC-Platform/pull/13714/files

I think that was then superseded by @afishbeck change in https://github.com/hpcc-systems/HPCC-Platform/pull/15016/files

Reply:
Probably that PR of Tony's should have removed defaultEsp support (and references to it in values.yaml / _helper.tpl then.

Just seems a bit inconsistent to allow the esp to use to be overridden in some cases but not others.

Reply:
https://github.com/hpcc-systems/HPCC-Platform/pull/15016/files was actually Jakes change.  But perhaps yes, I should have considered changing while adding mtls support.  I was trying to keep cert-manager support isolated from other changes as much as possible.